error: unexpected token
  --> $DIR/invalid.rs:12:15
   |
12 | #[pinned_drop(foo)] //~ ERROR unexpected token
   |               ^^^

error: #[pinned_drop] function must take a argument `Pin<&mut Type>`
  --> $DIR/invalid.rs:22:12
   |
22 | fn do_drop(_x: &mut Bar) {} //~ ERROR #[pinned_drop] function must take a argument `Pin<&mut Type>`
   |            ^^^^^^^^^^^^

error: aborting due to 2 previous errors

