error: #[pinned_drop] function must return the unit type
  --> $DIR/return-type.rs:13:32
   |
13 | fn do_drop(_x: Pin<&mut A>) -> bool {} //~ ERROR #[pinned_drop] function must return the unit type
   |                                ^^^^

error: #[pinned_drop] function must return the unit type
  --> $DIR/return-type.rs:22:32
   |
22 | fn do_drop(_x: Pin<&mut B>) -> ((),) {} //~ ERROR #[pinned_drop] function must return the unit type
   |                                ^^^^^

error: aborting due to 2 previous errors

