error: unexpected token
 --> $DIR/invalid.rs:7:10
  |
7 |     #[pin()] //~ ERROR unexpected token
  |          ^

error: unexpected token
  --> $DIR/invalid.rs:12:18
   |
12 | struct B<T>(#[pin(foo)] T); //~ ERROR unexpected token
   |                  ^

error: unexpected token
  --> $DIR/invalid.rs:16:12
   |
16 |     A(#[pin(foo)] T), //~ ERROR unexpected token
   |            ^

error: unexpected token
  --> $DIR/invalid.rs:22:14
   |
22 |         #[pin(foo)] //~ ERROR unexpected token
   |              ^

error: expected identifier
  --> $DIR/invalid.rs:27:27
   |
27 | #[pin_project(UnsafeUnpin,,)] //~ ERROR expected identifier
   |                           ^

error: aborting due to 5 previous errors

