error: #[pin_project] attribute may not be used on #[repr(packed)] types
 --> $DIR/packed.rs:6:8
  |
6 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] type
  |        ^^^^^^

error: #[pin_project] attribute may not be used on #[repr(packed)] types
  --> $DIR/packed.rs:13:8
   |
13 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] type
   |        ^^^^^^

error: aborting due to 2 previous errors

