error: pin_projectable may not be used on #[repr(packed)] types
 --> $DIR/packed.rs:8:8
  |
8 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] type
  |        ^^^^^^

error: pin_projectable may not be used on #[repr(packed)] types
  --> $DIR/packed.rs:15:8
   |
15 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] type
   |        ^^^^^^

error: pin_projectable may not be used on #[repr(packed)] types
  --> $DIR/packed.rs:23:8
   |
23 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] type
   |        ^^^^^^

error: pin_projectable may not be used on #[repr(packed)] types
  --> $DIR/packed.rs:28:8
   |
28 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] type
   |        ^^^^^^

error: aborting due to 4 previous errors

