error: pin_projectable may not be used on #[repr(packed)] types
 --> $DIR/packed.rs:9:8
  |
9 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] type
  |        ^^^^^^

error: pin_projectable may not be used on #[repr(packed)] types
  --> $DIR/packed.rs:16:8
   |
16 | #[repr(packed, C)] //~ ERROR may not be used on #[repr(packed)] type
   |        ^^^^^^

error: aborting due to 2 previous errors

